Marlin/src/gcode/calibrate/G28.cpp

```diff
@@ -626,6 +625,8 @@ void GcodeSuite::G28() {
       safe_delay(SENSORLESS_STALLGUARD_DELAY); // Short delay needed to settle
     #endif
   #endif // HAS_HOMING_CURRENT
+  
+  if (doZ) do_move_after_z_homing();
```

Comment:
The move after homing is certainly intended for any situation where `G28` / `G28 Z` is happening, but is it universally appropriate for situations where Z homes first, and for the Foam Cutter?

I'd like to do a quick review of the code history to understand why it was limited to doing the final move where it was.

e.g., Is it safe to assume that setups that home Z to max and have`G28 Z` as the slicer profile End G-code, the final move to a lower position is disabled in the configuration.

If needed, we can clear `doZ` before we get here for any situations determined inappropriate.